frontend/src/hooks/useModal.ts

```diff
@@ -0,0 +1,10 @@
+import { useContext } from 'react';
+import { ModalContext } from '../components/modals/ModalContext';
+
+export const useModal = () => {
+  const context = useContext(ModalContext);
+  if (!context) {
+    throw new Error('useModal must be used within a ModalProvider');
+  }
+  return context;
+};
```

Comment:
no need for this check. Ideally you should place `provider` to top level of components tree.

Reply:
done